### PR TITLE
tests: spi_loopback: kinetis dspi fix

### LINKED ISSUE #77010: tests: spi_loopback: kinetis: dma: some test is not supported well in kinetis dspi
**Describe the bug**

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 --> frdm_k64f
 - What have you tried to diagnose or workaround this issue?
 - -> change the settings
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 --> the failure is there since those new test case added. as I do not enable fixture for spi_loopback, this issue is hidden in CI before

**To Reproduce**

```
west build -b frdm_k64f -p always -- -DOVERLAY_CONFIG="overlay-mcux-dspi-dma.conf" -DDTC_OVERLAY_FILE="overlay-mcux-dspi-dma.overlay"
west flash
```

see error

**Expected behavior**

test pass

**Impact**

SPI tests

**Logs and console output**

```
*** Booting Zephyr OS build v3.7.0-825-ga747d1a747ee ***
Running TESTSUITE spi_loopback
===================================================================
START - test_spi_loopback
I: SPI test on buffers TX/RX 0x200044a0/0x20004480, frame size = 8, DMA enabled (without CONFIG_NOCACHE_MEMORY)
I: SPI test slow config
I: Start complete multiple
I: Passed
I: Start complete loop
I: Passed
I: Start null tx
I: Passed
I: Start half start
I: Passed
I: Start half end
I: Passed
I: Skip every 4
I: Start rx bigger than tx
E: Timeout waiting for transfer complete
E: Code -116
Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/spi/spi_loopback/src/spi.c:491: spi_rx_bigger_than_tx: (ret is true)
SPI transceive failed
FAIL - test_spi_loopback in 0.250 seconds
===================================================================
TESTSUITE spi_loopback failed.
------ TESTSUITE SUMMARY START ------
SUITE FAIL -   0.00% [spi_loopback]: pass = 0, fail = 1, skip = 0, total = 1 duration = 0.250 seconds
- FAIL - [spi_loopback.test_spi_loopback] duration = 0.250 seconds
------ TESTSUITE SUMMARY END ------
===================================================================
RunID: be16631211227b52ce8b39482af4c524
PROJECT EXECUTION FAILED
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.7.0-825-ga747d1a747ee

### PR DESCRIPTION
**Pull request by hakehuang**
1. kenistis dspi does support rx and tx with different size, so skip the tx_bigger test.
2. dspi dma need internal alignment and 8192 bytes are too many for k64, so reduce to 1440 bytes

Fixes: #77010

### CODE REVIEW

**Comment by nashif** [comment]
that is not a hotfix, please refer to the hotfix description

**Comment by hakehuang** [comment]
@teburd , @tbursztyka  can you help to review, thanks

File: tests/drivers/spi/spi_loopback/src/spi.c [hunk 1]
```diff
@@ -58,7 +58,14 @@ static struct spi_dt_spec spi_slow = SPI_DT_SPEC_GET(SPI_SLOW_DEV, SPI_OP(FRAME_
 #define STACK_SIZE (512 + CONFIG_TEST_EXTRA_STACK_SIZE)
 #define BUF_SIZE 18
 #define BUF2_SIZE 36
+
+#ifdef CONFIG_DSPI_MCUX_EDMA
+/*DSPI DMA need aligned buffer for internal*/
+#define BUF3_SIZE 1440
```
Comment (tbursztyka) [inline tests/drivers/spi/spi_loopback/src/spi.c:64]: 1440 or 2440 (see your commit message) ?

Comment (hakehuang) [reply tests/drivers/spi/spi_loopback/src/spi.c:64]: 1440 typo

Comment (tbursztyka) [reply tests/drivers/spi/spi_loopback/src/spi.c:64]: ok. Btw in your commit message: s/fixing:/Fixes:

**Review by tbursztyka — APPROVED** [review]

**Comment by hakehuang** [comment]
@teburd  please review

**Comment by hakehuang** [comment]
@teburd please review

**Comment by hakehuang** [comment]
@teburd please help to take a look. Thanks

**Review by teburd — APPROVED** [review]

**Pull request merged**